### ISSUE TITLE
Too many calls at oban_peers table

### ISSUE THREAD

**Issue by lucas-cegatti** [sorentwo/oban#651]
### Environment

- Oban Version: 2.11
- PostgreSQL Version: 13.3
- Elixir & Erlang/OTP Versions (`elixir --version`): Elixir 1.13.1 OTP 24

### Current Behavior

Hello, thanks for the amazing library :D

We have just upgraded our application to Oban 2.11 and together we've also changed our notifier to `Oban.Notifiers.PG`, we use k8s and our nodes are connected through libcluster. At this environment we have a total of 10 nodes connected, some of them are apis and some workers (where the jobs are processed).

The problem is that we are experience an unusual amount of calls on the new `oban_peers` table, for instance, in a 1h window the insert and delete operations to that table were called 2.5 million times each, here's a metric exported from our observer tool:

![image](https://user-images.githubusercontent.com/4501172/155414973-e3521260-7435-4d82-b440-f03960cdde3a.png)

We tried to find correlations to the recent app changes but couldn't find any, we are experience the usual amount of request at that period of time, we hope this issue could help us find anything on what could be possibly happening.

Our configurations are:

```elixir
config :our_app, Oban,
  engine: Oban.Pro.Queue.SmartEngine,
  repo: OurApp.Repo,
  notifier: Oban.Notifiers.PG,
  queues: [
    queue_1: 32,
    queue_2: 32
  ],
  plugins: [
    {Oban.Pro.Plugins.DynamicPruner,
     state_overrides: [
       completed: {:max_len, 10_000},
       cancelled: {:max_age, {1, :day}},
       discarded: {:max_age, {12, :month}}
     ]},
    {Oban.Pro.Plugins.DynamicCron,
     [config]},
    Oban.Pro.Plugins.Lifeline,
    Oban.Pro.Plugins.Reprioritizer
  ]
```

### Expected Behavior

We don't know if that number of calls are a normal behaviour, we expected a small amount of calls to the `oban_peers` table but could be wrong at that assumption.

**Comment by sorentwo** [comment]
@lucas-cegatti Thanks for opening this issue. You have the correct assumption and this is definitely a bug! There have been some other discussions in Slack, and I'm working to diagnose the problem.

Can you share a little more information?

1. How is your database hosted? RDS, Cloud SQL, etc?
2. How many nodes are you running in?
3. Is this reproducible in your staging environment?

There is an alternate branch without trapped exits or rescues here: https://github.com/sorentwo/oban/tree/spike/peer-without-safety. It would be really helpful if you could try running that in a staging environment and report any exceptions or exits that you encounter.

**Comment by lucas-cegatti** [comment]
1. How is your database hosted? RDS, Cloud SQL, etc?
It's on RDS

3. How many nodes are you running in?
Single Node

5. Is this reproducible in your staging environment?
Fortunately it was at our staging environment, it didn't reach production

During this unusual amount of call our dev, [@arkanttus](https://github.com/arkanttus), found that the leader node was changing constantly at the table almost like it couldn't decide for one. We thought this could be a useful information.

We're changing our version to the branch you've indicated, we'll let you know the results.

Thanks!!

**Comment by lucas-cegatti** [comment]
@sorentwo sorry, at this question you meant database or k8s nodes? 

`How many nodes are you running in?`

**Comment by sorentwo** [comment]
@lucas-cegatti For that question, I meant k8s nodes, the nodes running Oban.

**Comment by lucas-cegatti** [comment]
> @lucas-cegatti For that question, I meant k8s nodes, the nodes running Oban.

3 nodes

**Comment by sorentwo** [comment]
I've found the issue. Fix and a new release are incoming.

**Comment by sorentwo** [comment]
Fix is out in Oban v2.11.2

**Comment by lucas-cegatti** [comment]
Thank you very much, @sorentwo
